### PR TITLE
fix to allow an annotations to appear multiple times with differnt pr…

### PR DESCRIPTION
**Pull request by galiri**
…edicates on read

### CODE REVIEW

File: annotations/augmenter_test.go [hunk 1]
```diff
@@ -94,11 +94,63 @@ var expectedAugmentedAnnotations = []Annotation{
...
+		ConceptId: "http://www.ft.com/thing/b224ad07-c818-3ad6-94af-a4d351dbb619",
+	},
+	{
+		Predicate: "http://www.ft.com/ontology/hasDisplayTag",
+		ConceptId: "http://www.ft.com/thing/b224ad07-c818-3ad6-94af-a4d351dbb619",
+	},
+
+}
+
+var expectedMulitPredicateAugmentedAnnotations = []Annotation{
```
Comment (khatton-ft) [inline annotations/augmenter_test.go]: :pencil2: Mul_ti_

File: annotations/augmenter_test.go [hunk 2]
```diff
@@ -94,11 +94,63 @@ var expectedAugmentedAnnotations = []Annotation{
...
+		IsFTAuthor: false,
+	},
+
+}
+
+var testMultiPredicateConceptIDs = []string{
+	"b224ad07-c818-3ad6-94af-a4d351dbb619",
+}
+
+var testMultiPredicteConcept= map[string]concept.Concept{
```
Comment (khatton-ft) [inline annotations/augmenter_test.go]: :pencil2: Predic_a_te

File: annotations/augmenter_test.go [hunk 3]
```diff
@@ -113,11 +165,38 @@ func TestAugmentAnnotations(t *testing.T) {
...
+	ctx := tidUtils.TransactionAwareContext(context.Background(), tidUtils.NewTransactionID())
+	conceptsSearchAPI.
+		On("SearchConcepts", ctx, matcher).
+		Return(testMultiPredicteConcept, nil)
+	a := NewAugmenter(conceptsSearchAPI)
+
+	annotations, err := a.AugmentAnnotations(ctx, testMultiCanonicalizedAnnotations)
+
+	assert.NoError(t, err)
+	assert.Equal(t, len(expectedMulitPredicateAugmentedAnnotations), len(annotations))
```
Comment (khatton-ft) [inline annotations/augmenter_test.go]: So in fact, L184-187 could be replaced by `assert.ElementsMatch`

File: annotations/augmenter_test.go [hunk 4]
```diff
@@ -113,11 +165,38 @@ func TestAugmentAnnotations(t *testing.T) {
 	conceptsSearchAPI.AssertExpectations(t)
 }
 
+
+func TestAugmentAnnotationsMultiPredicatePerConcept(t *testing.T) {
+	matcher := mock.MatchedBy(func(l1 []string) bool {
+		return assert.ElementsMatch(t, l1,testMultiPredicateConceptIDs)
```
Comment (khatton-ft) [inline annotations/augmenter_test.go]: :+1: ... though you have to read to godoc to be certain the order of elements isn't significant.

File: annotations/augmenter.go [hunk 5]
```diff
@@ -63,17 +64,23 @@ func (a *annotationAugmenter) AugmentAnnotations(ctx context.Context, canonicalA
...
 	for _, ann := range canonicalAnnotations {
 		conceptUUID := extractUUID(ann.ConceptId)
 		if conceptUUID != "" {
-			keys = append(keys, conceptUUID)
-			mappedConcepts[conceptUUID] = ann
+			conceptUUIDs[conceptUUID] = empty
 		}
 	}
-	return keys, mappedConcepts
+	for k,_:= range conceptUUIDs{
```
Comment (izzyblues) [inline annotations/augmenter.go]: Forgot `go fmt` ;)

Comment (izzyblues) [reply annotations/augmenter.go]: And you can re write it as `for k := range conceptUUIDs {`

**Pull request merged**